pkg/registration/spoke/spokeagent.go

```diff
@@ -226,33 +206,31 @@ func (o *SpokeAgentConfig) RunSpokeAgentWithSpokeInformers(ctx context.Context,
 		o.currentBootstrapKubeConfig = o.registrationOption.BootstrapKubeconfig
 	}
 
-	// load bootstrap client config and create bootstrap clients
-	bootstrapClientConfig, err := clientcmd.BuildConfigFromFlags("", o.currentBootstrapKubeConfig)
-	if err != nil {
-		return fmt.Errorf("unable to load bootstrap kubeconfig from file %q: %w", o.currentBootstrapKubeConfig, err)
-	}
-	bootstrapKubeClient, err := kubernetes.NewForConfig(bootstrapClientConfig)
+	kubeconfig, err := clientcmd.LoadFromFile(o.currentBootstrapKubeConfig)
 	if err != nil {
 		return err
 	}
-	bootstrapClusterClient, err := clusterv1client.NewForConfig(bootstrapClientConfig)
+
+	// build up the secretOption
+	secretOption := register.SecretOption{
+		SecretNamespace:          o.agentOptions.ComponentNamespace,
+		SecretName:               o.registrationOption.HubKubeconfigSecret,
+		ClusterName:              o.agentOptions.SpokeClusterName,
+		AgentName:                o.agentOptions.AgentID,
+		ManagementSecretInformer: namespacedManagementKubeInformerFactory.Core().V1().Secrets().Informer(),
+		ManagementCoreClient:     managementKubeClient.CoreV1(),
+		HubKubeconfigFile:        o.agentOptions.HubKubeconfigFile,
+		HubKubeconfigDir:         o.agentOptions.HubKubeconfigDir,
+		BootStrapKubeConfig:      kubeconfig,
```

Comment:
could we use `o.currentBootstrapKubeConfig` instead of loading the kubeconfig here?

Reply:
this is loaded from currentBootstrapKubeConfig

Reply:
I mean we put the BootstrapKubeConfig file path in the SecretOption, not a kubeconfig object, we load the kubeconfig object in the BuildClientsFromSecretOption
in feature, if we don't use kube bootstrap, we can handle this in one place, does it make sense?

Reply:
I see, yeah we could do that.

Reply:
OK, got it